### PR TITLE
UCSC Rest migration

### PR DESCRIPTION
**Pull request by sanchit-saini**

### CODE REVIEW

**Comment by lawremi** [comment]
Are you still planning to access tracks through the track hub? At least the WIG/BigWig ones?

File: R/trackhub.R [hunk 1]
```diff
@@ -692,48 +692,55 @@ createTrack <- function(trackDf) {
...
-    fileContent <- gsub(",", ";", fileContent)
-    contentDf <- read.csv(text = sub(" ", ",", fileContent), header = FALSE)
-    contentDf$V2 <- gsub(";", ",", contentDf$V2)
-    nonEmptyContent <- vapply(contentDf$V2, function(x) x!="", logical(1L))
-    contentDf <- contentDf[nonEmptyContent,]
-    contentDf
+    regex <- "^\\s*\\t*([a-zA-Z.]+)\\s?(.*)$"
+    field <- sub(regex, "\\1", fileContent)
+    value <- sub(regex, "\\2", fileContent)
+    contentDf <- data.frame(field, value)
```
Comment (lawremi) [inline R/trackhub.R:722]: Are you sure `read.table()` cannot do this?

File: R/ranges.R [hunk 2]
```diff
@@ -196,6 +196,37 @@ singleGenome <- function(x) {
 }
 
 
+seqlengthsFromAPI <- function(genome) {
```
Comment (lawremi) [inline R/ranges.R:199]: As we discussed, all of these helper functions can probably go away.

File: R/trackhub.R (hunk 1 above)
Comment (sanchit-saini) [reply R/trackhub.R:722]: Yes, `read.table()` was able to parse the file but it misinterpreted part of the URL(with `#`) as a comment and omit some part of the URL string.

```R
content <- read.table("https://ftp.ncbi.nlm.nih.gov/snp/population_frequency/TrackHub/20200227123210/hg19/trackDb.txt", strip.white=T, header = F, sep=" ", fill = T)
content[23,]

    V1                                  V2 V3 V4 V5
23 url https://www.ncbi.nlm.nih.gov/snp/$$

```
Expected URL string
```R
url https://www.ncbi.nlm.nih.gov/snp/$$#frequency_tab
```


Along with that, I called `read.table()` with `fill` as True to make it work and it created a data frame with multiple columns, these columns need to be transformed into two column(i.e as a key-value pair).

Transformation of a data frame is possible somewhat I like this

```R
content <- read.table("https://ftp.ncbi.nlm.nih.gov/snp/population_frequency/TrackHub/20200227123210/hg19/trackDb.txt", strip.white=T, header = F, sep=" ", fill = T)
colNames = setdiff(colnames(content), "V1")
content$V2 = apply(content[, colNames, drop = F], MARGIN = 1, FUN = function(i) paste(i, collapse = " "))
content[setdiff(colNames, "V2")] <- NULL
```


but I don't know How should I handle the unintended omission of the content safely?
One way I can think of to fix this is to set the comment character(`comment.char`) to some other character in `read.table()` not sure whether it is safe or not.

File: R/ranges.R (hunk 2 above)
Comment (sanchit-saini) [reply R/ranges.R:199]: Yes, I'll update and push the changes.

File: R/trackhub.R (hunk 1 above)
Comment (lawremi) [reply R/trackhub.R:722]: I think it's worth playing with the `comment.char=` argument, as I doubt they are using comments in these track files (would they even be valid?).

Also, I think you can do the paste in a simpler way, like:
```
content$V2 <- do.call(paste, c(content[, colNames], sep=" "))
```

Comment (sanchit-saini) [reply R/trackhub.R:722]: Okay, I will update accordingly.
Yes, the tracked file can contain comments and it treated as a valid file.

**Pull request merged**